src/components/createElementComponent.tsx

```diff
@@ -138,21 +139,24 @@ const createElementComponent = (
     useAttachEvent(element, 'checkout', checkoutCallback);
 
     React.useLayoutEffect(() => {
-      if (element === null && elements && domNode.current !== null) {
+      if (elementRef.current === null && elements && domNode.current !== null) {
         const newElement = elements.create(type as any, options);
         if (type === 'cart' && setCart) {
           // we know that elements.create return value must be of type StripeCartElement if type is 'cart',
           // we need to cast because typescript is not able to infer which overloaded method is used based off param type
           setCart((newElement as unknown) as stripeJs.StripeCartElement);
         }
+
+        elementRef.current = newElement;
         setElement(newElement);
```

Comment:
Add comments, otherwise this feels redundant at a first glance!
```suggestion
        // Store element in a ref to ensure it's _immediately_ available in cleanup hooks in StrictMode
        elementRef.current = newElement;
        // Store element in state to to facilitate event listener attachment
        setElement(newElement);
```

Reply:
Good call!